extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +84,48 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+    function loadHQImage_() {
+      // We only need to handle error once to load another source.
+      imgPlaceholder.removeEventListener('error', loadHQImage_);
+      imgPlaceholder.setAttribute('src', 'https://i.ytimg.com/vi/' +
+          videoid + '/hqdefault.jpg');
+    }
+
+    function maybeLoadHQImage_() {
```

Comment:
I think this code can be rewritten a bit clearer if using `loadPromise` and promise chaining.